core/src/main/java/com/datastrato/gravitino/catalog/PartitionDispatcher.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.catalog;
+
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.exceptions.NoSuchPartitionException;
+import com.datastrato.gravitino.exceptions.PartitionAlreadyExistsException;
+import com.datastrato.gravitino.rel.SupportsPartitions;
+import com.datastrato.gravitino.rel.partitions.Partition;
+
+/**
+ * {@code PartitionDispatcher} interface is a wrapper around the {@link SupportsPartitions}
+ * interface, adding {@link NameIdentifier} of table to the method parameters for find out the
+ * catalog class loader.
+ */
+public interface PartitionDispatcher {
+
+  /**
+   * List the names of all partitions in the table.
+   *
+   * @param tableIdent The identifier of the table.
+   * @return The names of all partitions in the table.
+   */
+  String[] listPartitionNames(NameIdentifier tableIdent);
+
+  /**
+   * List all partitions in the table.
+   *
+   * @param tableIdent The identifier of the table.
+   * @return The list of partitions.
+   */
+  Partition[] listPartitions(NameIdentifier tableIdent);
+
+  /**
+   * Get a partition by name from the table.
+   *
+   * @param tableIdent The identifier of the table.
+   * @param partitionName The name of the partition.
+   * @return The partition.
+   * @throws NoSuchPartitionException
+   */
+  Partition getPartition(NameIdentifier tableIdent, String partitionName)
+      throws NoSuchPartitionException;
+
+  /**
+   * Check if a partition exists in the table.
+   *
+   * @param tableIdent The identifier of the table.
+   * @param partitionName The name of the partition.
+   * @return True if the partition exists, false otherwise.
+   */
+  default boolean partitionExists(NameIdentifier tableIdent, String partitionName) {
+    try {
+      getPartition(tableIdent, partitionName);
+      return true;
+    } catch (Exception e) {
+      return false;
```

Comment:
Is it acceptable to directly return false if any exceptions occur? What if the partition exists but we can't get it temporarily, such as due to IO throttling or network congestion?

Reply:
fixed